File: javaCore/src/main/java/com/hamdan/forzenbook/java/core/utils/DialogUtils.java
package com.hamdan.forzenbook.java.core.utils;

import android.app.AlertDialog;
import android.app.DatePickerDialog;
import android.content.Context;
import android.content.DialogInterface;

import com.hamdan.forzenbook.java.core.GeneralUtilityFunctions;

import java.util.Calendar;

public class DialogUtils {

    public static AlertDialog standardAlertDialog(Context context, String title, String body, String buttonText, Runnable onDismiss) {
        AlertDialog dialog = new AlertDialog.Builder(context).create();
        dialog.setTitle(title);
        dialog.setMessage(body);
        dialog.setButton(DialogInterface.BUTTON_POSITIVE, buttonText, (dialogInterface, i) -> dialog.dismiss());
        dialog.setOnDismissListener(listener -> onDismiss.run());
        dialog.show();
        return dialog;
    }

    public static void datePickerDialog(Context context, String birthDate, DateDialogOnTextChange onTextChange, Runnable onDateSubmission, Runnable onDismiss) {
        Calendar calendar = Calendar.getInstance();
        int currentYear = calendar.get(Calendar.YEAR);
        int currentMonth = calendar.get(Calendar.MONTH);
        int currentDay = calendar.get(Calendar.DAY_OF_MONTH);
        int selectedYear = currentYear;
        int selectedMonth = currentMonth;
        int selectedDay = currentDay;
        if (!birthDate.isEmpty()) {
            String[] split = birthDate.split("-");
            if(split.length==3){
                selectedYear = Integer.parseInt(split[2]);
                selectedMonth = Integer.parseInt(split[0]) - 1;
                selectedDay = Integer.parseInt(split[1]);
            }
        }
        DatePickerDialog.OnDateSetListener dateSetListener;
        dateSetListener = (view, year, month, day) -> {
            String birthDay = GeneralUtilityFunctions.stringDate(month + 1, day, year, context);
            onTextChange.operation(birthDay);
            onDateSubmission.run();
        };
        DatePickerDialog dialog = new DatePickerDialog(
                context,
                com.hamdan.forzenbook.ui.core.R.style.MySpinnerDatePickerStyle,
                dateSetListener,
                selectedYear,
                selectedMonth,
                selectedDay
        );
        dialog.setOnDismissListener(dialogInterface -> {
            onDismiss.run();
        });
        dialog.show();
    }


}

File: legacyCore/src/main/java/com/hamdan/forzenbook/legacy/core/view/utils/DialogUtils.kt
package com.hamdan.forzenbook.legacy.core.view.utils

import android.app.AlertDialog
import android.app.DatePickerDialog
import android.content.Context
import android.widget.DatePicker
import com.hamdan.forzenbook.core.stringDate
import com.hamdan.forzenbook.legacy.core.view.FragmentAlertDialog
import com.hamdan.forzenbook.legacy.core.view.FragmentDatePickerDialog
import com.hamdan.forzenbook.ui.core.R
import java.util.Calendar

object DialogUtils {
    fun standardAlertDialog(
        context: Context,
        title: String,
        body: String,
        buttonText: String,
        onDismiss: () -> Unit,
    ): AlertDialog {
        val dialog = AlertDialog.Builder(context).apply {
            setTitle(title)
            setMessage(body)
            setPositiveButton(buttonText) { dialog, _ ->
                dialog.dismiss()
            }
        }.create().apply {
            setOnDismissListener {
                onDismiss()
            }
        }
        dialog.show()
        return dialog
    }

    fun fragmentAlertDialog(
        title: String,
        body: String,
        buttonText: String,
        onDismiss: () -> Unit,
    ): FragmentAlertDialog = FragmentAlertDialog(
        title, body, buttonText, onDismiss,
    )

    fun fragmentDatePicker(
        birthDate: String,
        onTextChange: (String) -> Unit,
        onDateSubmission: () -> Unit,
        onDateDismiss: () -> Unit
    ): FragmentDatePickerDialog = FragmentDatePickerDialog(
        birthDate, onTextChange, onDateSubmission, onDateDismiss
    )
}
